pages/in/settings/others.vue

```diff
@@ -90,11 +152,39 @@ export default {
     disabled() {
       return this.isLoading || !this.getUserIsRegistered;
     },
-    hasChanged() {
-      const user = this.getUserInfo;
-      return this.getUserIsRegistered && (
-        this.isEmailEnabled !== user.isEmailEnabled
-      );
+    actualPaymentRedirectUrlWhitelist() {
+      return Array.from(new Set(this.paymentRedirectUrlWhitelist.map(obj => obj.url)
+        .filter(url => !!url)));
+    },
+    allUrlsValid() {
+      return this.paymentRedirectUrlWhitelist.every(obj => !obj.err);
+    },
+  },
+  watch: {
+    paymentRedirectUrlWhitelist: {
+      handler(newValue) {
+        const urlArr = newValue.map(obj => obj.url);
+        newValue.forEach((obj, i) => {
+          if (obj.url) {
+            if (!isValidHttpUrl(obj.url)) {
+              obj.err = 'errInvalidUrl';
+            } else if (urlArr.indexOf(obj.url) !== i) {
+              obj.err = 'errDuplicatedUrl';
+            } else {
+              obj.err = '';
+            }
+          } else {
+            obj.err = '';
+          }
+        });
+        const urlNum = newValue.length;
+        if (urlNum === 0 || (newValue[urlNum - 1].url !== '' && urlNum < MAX_REDIRECT_URL_NUM)) {
+          newValue.push({ url: '', err: '' });
+        } else if (urlNum > 1 && newValue[urlNum - 1].url === '' && newValue[urlNum - 2].url === '') {
```

Comment:
instead of comparing with empty string, prefer checking falsey `!(newValue[urlNum - 1].url )` , etc